src/realm/object-store/impl/transact_log_handler.cpp

```diff
@@ -76,16 +76,18 @@ KVOAdapter::KVOAdapter(std::vector<BindingContext::ObserverState>& observers, Bi
     for (auto& observer : observers) {
         auto table = group.get_table(TableKey(observer.table_key));
         for (auto key : table->get_column_keys()) {
-            if (table->get_column_attr(key).test(col_attr_List))
-                m_lists.push_back({&observer, {}, key});
+            if (table->get_column_attr(key).test(col_attr_List)) {
+                auto obj = table->get_object(observer.obj_key);
+                m_lists.push_back({&observer, {}, obj.get_path().path_from_top});
```

Comment:
Not so sure about this change. I would assume you should just replace `key` with `{key}`

Reply:
I thought `ListInfo` would need to have the Path instead of the column key.